lib/src/tools/format_tool.dart

```diff
@@ -366,6 +373,42 @@ Iterable<String> buildArgs(
   return args;
 }
 
+/// Builds and returns the full list of args for the formatter process that
+/// [FormatTool] will start.
+///
+/// [executableArgs] will be included first and will include the
+/// `format` executable (e.g. `dart format`).
+///
+/// Next, [mode] will be mapped to the appropriate formatter arg(s), e.g. `-o`,
+/// and included.
+///
+/// If non-null, [configuredFormatterArgs] will be included next.
+///
+/// If [argResults] is non-null and the `--formatter-args` option is non-null,
+/// they will be included next.
+///
+/// Finally, if [verbose] is true and the verbose flag (`-v`) is not already
+/// included, it will be added.
+Iterable<String> buildArgsForDartFormat(
+    Iterable<String> executableArgs, FormatMode mode,
+    {ArgResults argResults, List<String> configuredFormatterArgs}) {
+  final args = <String>[
+    ...executableArgs,
+
+    // Combine all args that should be passed through to the dart format in this
+    // order:
+    // 1. Mode flag(s), if configured
+    if (mode == FormatMode.check) '--set-exit-if-changed',
+    if (mode == FormatMode.dryRun) ...['-o', 'none', '.'],
```

Comment:
The `.` at the end shouldn't be included here since we add the files/dirs to format below on line 525
```suggestion
    if (mode == FormatMode.dryRun) ...['-o', 'none'],
```